redis/tests/support/mod.rs

```diff
@@ -244,6 +244,13 @@ impl RedisServer {
         modules: &[Module],
         spawner: F,
     ) -> RedisServer {
+        #[cfg(feature = "rustls")]
+        if rustls::crypto::CryptoProvider::get_default().is_none() {
+            rustls::crypto::ring::default_provider()
```

Comment:
The `ring` provider isn't enabled by default in rustls 0.23, because the default switched to aws-lc-rs. You can probably just use the aws-lc-rs default provider for this tests, but as a downstream user I'd like a way to disable the aws-lc-rs provider (which is enabled via rustls' default features) if I want to use ring instead.

Reply:
> but as a downstream user I'd like a way to disable the aws-lc-rs provider (which is enabled via rustls' default features) if I want to use ring instead.

how could we do that? I'd like to avoid the addition of new features.

Reply:
do you suggest avoiding pulling the default features?

Reply:
> > but as a downstream user I'd like a way to disable the aws-lc-rs provider (which is enabled via rustls' default features) if I want to use ring instead.
> 
> how could we do that? I'd like to avoid the addition of new features.

At the very least you should depend on rustls (and tokio-rustls) with `default-features = false`, yes. But then if you want to avoid panics you probably also want to have features that do enable one of the crypto providers.

Reply:
so, that's a backwards incompatible change? I'll leave it out of this PR, then.

Reply:
What exactly do you see as a backwards incompatible change? I think you can still make this all work out without backwards incompatible changes.

Reply:
Correct me if I'm wrong here, but if we add `default-features = false`, then we can't initialize a crypto provider here
https://github.com/redis-rs/redis-rs/blob/3cf579fd0da327d018559f1981e6cc27d6fe9cb3/redis/src/connection.rs#L906

and so this path will panic for a user that didn't initialize a crypto provider beforehand, right?

Reply:
You'd add `default-features = false` but also `features = [..]` to enable the some of the default features and pick a crypto provider.